packages/react-ui-core/src/Highlighter/Highlighter.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import { randomId } from '@rentpath/react-ui-utils'
+
+@themed(/^Highlighter/)
+
+export default class Highlighter extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    pattern: PropTypes.string.isRequired,
+    children: PropTypes.string,
+    theme: PropTypes.object,
+    indexHighlighted: PropTypes.number,
+  }
+
+  static defaultProps = {
+    text: '',
+    children: '',
+    theme: {},
+  }
+
+  get indexHighlighted() {
+    return this.props.indexHighlighted + 1
+  }
+
+  renderHighlightedText() {
+    const { pattern, children } = this.props
+
+    if (!pattern) return children
+
+    const segments = children.split(pattern)
+
+    return segments.reduce((prev, curr, index) => [prev,
+      this.renderHighlightedSegment(index),
+      curr])
+  }
+
+  renderHighlightedSegment(index) {
+    const {
+      pattern,
+      theme,
+      indexHighlighted,
+      className,
+    } = this.props
+
+    if (!indexHighlighted || index === indexHighlighted) {
+      return (
+        <span
+          data-tid="highlighter-match"
```

Comment:
Since this will be repeated, we may want to make the data-tid specific, but I'm not sure. Maybe:
```
data-tid=`highlighter-match-${index}`
```
I'm not sure if we _really_ need this, so please give thoughts on this.

Reply:
@yasabere You a little more familiar with this code. What do you think?